pkg/config/stepmeta.go

```diff
@@ -302,12 +310,43 @@ func (m *StepData) GetContextDefaults(stepName string) (io.ReadCloser, error) {
 	}
 
 	if len(m.Spec.Sidecars) > 0 {
-		if len(m.Spec.Sidecars[0].Command) > 0 {
-			root["sidecarCommand"] = m.Spec.Sidecars[0].Command[0]
-		}
-		m.Spec.Sidecars[0].commonConfiguration("sidecar", &root)
-		putStringIfNotEmpty(root, "sidecarReadyCommand", m.Spec.Sidecars[0].ReadyCommand)
+		// if there one side car do not check conditions and consider the only side care as default . this is default behaviour
+		// if there are more than one side car then check conditions,
+		if len(m.Spec.Sidecars) == 1 {
```

Comment:
merge with outer `if

Reply:
That would slightly change the meaning, but looping over empty sidecars should be fine. But why do we need those `if`s at all? Why shouldn't one `sidecar` have a condition? Taking always the `else` case (i.e. looping over all sidecars) should always working, for 0,1 and many.